lib/sync-DataSetModel.js

```diff
@@ -406,25 +429,32 @@ var self = {
     self.doListHandler(dataset_id, datasetClient.queryParams, function (err, records) {
       if (err) return cb(err);
 
-
-      var hashes = [];
       var recOut = {};
-      for (var i in records) {
-        var rec = {};
-        var recData = records[i];
-        var hash = syncUtil.generateHash(recData);
-        hashes.push(hash);
-        rec.data = recData;
-        rec.hash = hash;
-        recOut[i] = rec;
-      }
-      var globalHash = syncUtil.generateHash(hashes);
+      var hashes = [];
+
+      async.forEachSeries(Object.keys(records), function(recordId, itemCallback) {
+        var record = records[recordId];
+        syncUtil.asyncGenerateHash(record, function(err, recordHash) {
+          var rec = {
+            data: record,
+            hash: recordHash
+          };
+          hashes.push(recordHash);
+          recOut[recordId] = rec;
+          itemCallback();
+        });
+      }, function(err) {
+        if (err) return cb(err);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)